go-controller/pkg/ovn/libovsdbops/router.go

```diff
@@ -195,7 +195,7 @@ func isEquivalentNAT(existing *nbdb.NAT, searched *nbdb.NAT) bool {
 	}
 
 	// Allow externalIP to be empty if type is SNAT.
-	if (searched.ExternalIP != "" || searched.Type != nbdb.NATTypeSNAT) &&
+	if (searched.ExternalIP != "" || searched.Type == nbdb.NATTypeSNAT) &&
```

Comment:
@flavio-fernandes : Does this tiny change make sense to you? I agree with your code comment on L197, but the match condition is doing the opposite. I ran into a problem when deleting the `dnat_and_snat` entries since we were returning false straightaway from here since externalIP was nil and so it didn't proceed further to match on the logicalPort case. Wanna make sure I don't change some other logic,... wdyt?

Reply:
hmm... I think the original code is what I intended, but I may have messed up dnat_and_snat.
The idea was that if 'searched.ExternalIP' should be checked when type is dnat_and_snat, unless
it is "" or if this was snat). It cannot be `nil`, right? Let's go over your case later this week. I may have
done something wrong here or a new clause may be needed for dnat_and_snat. Sorry for that!

Reply:
Now I'm thinking this should just be:
```
if searched.ExternalIP != "" && searched.ExternalIP != existing.ExternalIP {
   return false
}
```
😊

Reply:
yeah the condition you mentioned above makes more sense, there is no point in comparing `ExternalIP` if its an emtpy string :)